js/plugins/dotprompt/src/prompt.ts

```diff
@@ -166,9 +168,12 @@ export class Dotprompt<Variables = unknown> implements PromptMetadata {
     );
   }
 
-  private _generateOptions<O extends z.ZodTypeAny = z.ZodTypeAny>(
+  private _generateOptions<
+    O extends z.ZodTypeAny = z.ZodTypeAny,
+    CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
+  >(
     options: PromptGenerateOptions<Variables>
-  ): GenerateOptions<z.ZodTypeAny, O> {
+  ): GenerateOptions<O, CustomOptions> {
```

Comment:
Why is the ordering inconsistent? Sometimes its `<O, CustomOptions> `and other times its `<CustomOptions, O>`...

Reply:
Unfortunately, the signature for `GenerateOptions` is `GenerateOptions<O, CustomOptions>` and the signature for `generate` is `generate<CustomOptions, O>`. Since the dotprompt `generate()` method parallels the core generate function, I kept that signature consistent.

Reply:
Also, this line was a bug in the previous version. It worked most of the time but only for Anys.